SDK/include/NDK/Components/CameraComponent.hpp

```diff
@@ -33,20 +33,20 @@ namespace Ndk
 			inline void EnsureViewMatrixUpdate() const;
 			inline void EnsureViewportUpdate() const;
 
-			inline float GetAspectRatio() const;
-			inline Nz::Vector3f GetEyePosition() const;
-			inline Nz::Vector3f GetForward() const;
+			inline float GetAspectRatio() const override;
+			inline Nz::Vector3f GetEyePosition() const override;
+			inline Nz::Vector3f GetForward() const override;
 			inline float GetFOV() const;
-			inline const Nz::Frustumf& GetFrustum() const;
+			inline const Nz::Frustumf& GetFrustum() const override;
 			inline unsigned int GetLayer() const;
-			inline const Nz::Matrix4f& GetProjectionMatrix() const;
+			inline const Nz::Matrix4f& GetProjectionMatrix() const override;
 			inline Nz::ProjectionType GetProjectionType() const;
-			inline const Nz::RenderTarget* GetTarget() const;
+			inline const Nz::RenderTarget* GetTarget() const override;
 			inline const Nz::Rectf& GetTargetRegion() const;
-			inline const Nz::Matrix4f& GetViewMatrix() const;
-			inline const Nz::Recti& GetViewport() const;
-			inline float GetZFar() const;
-			inline float GetZNear() const;
+			inline const Nz::Matrix4f& GetViewMatrix() const override;
+			inline const Nz::Recti& GetViewport() const override;
+			inline float GetZFar() const override;
+			inline float GetZNear() const override;
```

Comment:
Not useless since it wouldn't compile without this keyword, but since it's
virtual the compiler will probably compile the methods and will do virtual
calls.

The thing to do is to remove the methods from the inline file and put them
in the source code .cpp
I'll do it if you want
Le 31 déc. 2015 2:08 PM, "Gawaboumga" <notifications@github.com> a écrit :

> In SDK/include/NDK/Components/CameraComponent.hpp
> <https://github.com/DigitalPulseSoftware/NazaraEngine/pull/51#discussion_r48656386>
> :
>
> >  			inline const Nz::Rectf& GetTargetRegion() const;
> > -			inline const Nz::Matrix4f& GetViewMatrix() const;
> > -			inline const Nz::Recti& GetViewport() const;
> > -			inline float GetZFar() const;
> > -			inline float GetZNear() const;
> > +			inline const Nz::Matrix4f& GetViewMatrix() const override;
> > +			inline const Nz::Recti& GetViewport() const override;
> > +			inline float GetZFar() const override;
> > +			inline float GetZNear() const override;
>
> I agree inline is useless in this context
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DigitalPulseSoftware/NazaraEngine/pull/51/files#r48656386>
> .
>